model/dao/ProjectDAO/PostgreSQLProjectDAO.php

```diff
@@ -472,13 +470,14 @@ public function getByCustomerUserLogin($customerId = NULL, $userLogin = NULL,
      *        field.
      * @param string $filterCname string to filter projects by their customer name. NULL
      *        to deactivate filtyering by this field
+     * @param boolean $filter flag to check if the request has filtering variables set
```

Comment:
I don't really like this parameter because it introduces an artificial and unexplained distinction among filtering variables, some would work even when `$filter` is false and some wouldn't.

You can check if the filtering variables you are interested are enabled or not, checking if they are null, and modify the query accordingly. For example, you may add the customer table only if `$filterCname != NULL`. Conditions, in general, will only add to the query time if the user has specified them.

Anyway, we actually need one parameter, that could be called `$returnExtendedInfo`, which would return all the dedication and cost information that comes from the extended query that involved the task table:
* If it is true, you will have to join the `task` and `hour_cost_history` tables, and add the `GROUP BY` clause.
* If it is false, you will have a simpler query without those two joins. It would default to false.